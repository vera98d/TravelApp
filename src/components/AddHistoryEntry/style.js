import styled, { keyframes } from "styled-components";

export const Form = styled.form`
  width: 100%;
  height: 90%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  margin-top: 1.5em;

  @media only screen and (orientation: landscape) and (max-width: 750px) {
    margin-top: 0.4em;
  }
`;

export const Input = styled.input`
  display: block;
  width: 100%;
  padding: 1em;
  margin-top: 1em;
  border: 0px solid white;
  border-radius: 15px;
  font-family: ${(props) => props.theme.fontFamily.secondaryFont};

  &:focus {
    outline: none;
    box-shadow: 0px 0px 2px rgba(191, 125, 101, 0.77);
  }

  @media only screen and (orientation: landscape) and (max-width: 750px) {
    margin-top: 0.7em;
    padding: 0.8em;
  }

  @media only screen and (orientation: landscape) and (max-height: 350px) {
    margin-top: 0.4em;
    padding: 0.5em;
  }
`;

export const TextArea = styled.textarea`
  display: block;
  width: 100%;
  height: 25em;
  padding: 1em;
  margin-top: 1.5em;
  border: 0px solid white;
  border-radius: 15px;
  resize: none;
  flex-grow: 2;
  font-family: ${(props) => props.theme.fontFamily.secondaryFont};

  &:focus {
    outline: none;
    box-shadow: 0px 0px 2px rgba(191, 125, 101, 0.77);
  }

  @media only screen and (orientation: landscape) and (max-width: 750px) {
    margin-top: 0.7em;
    padding: 0.8em;
  }

  @media only screen and (orientation: landscape) and (max-height: 350px) {
    margin-top: 0.4em;
    padding: 0.5em;
  }
`;

export const ButtonGroup = styled.div`
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-top: 1.5em;

  @media only screen and (orientation: landscape) and (max-width: 750px) {
    margin-top: 0.7em;
  }

  @media only screen and (orientation: landscape) and (max-height: 350px) {
    margin-top: 0.4em;
  }
`;

export const ErrorMessage = styled.p`
  color: #c23e3e;
  align-self: flex-start;
  margin: 0.5em 0 0 1em;
`;

export const AddImageButton = styled.img`
  width: 40px;
  height: 40px;
  cursor: pointer;

  &:hover {
    opacity: 0.8;
  }

  @media only screen and (orientation: landscape) and (max-width: 750px) {
    width: 30px;
    height: 30px;
    top: 0.4em;
    right: 1.5em;
  }
`;

export const FileInput = styled.input`
  display: none;
`;

export const ImagePreview = styled.img`
  max-width: 60px;
  max-height: 60px;
`;

const rotate360 = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

export const Spinner = styled.div`
  animation: ${rotate360} 1s linear infinite;
  transform: translateZ(0);

  border-top: 2px solid #dae5e5;
  border-right: 2px solid #dae5e5;
  border-bottom: 2px solid #b1e0d9;
  border-left: 4px solid #92d6ca;
  background: transparent;
  width: 130px;
  height: 130px;
  border-radius: 50%;
`;

const appear = keyframes`
  from {
    opacity:0;
  }
  to {
    opacity:100%;
  }
`;

export const FlexColContainer = styled.div`
  width: ${(props) => props.width};
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: ${(props) => props.align};
  justify-content: center;
  gap: ${(props) => props.gap};
  animation: ${appear} 0.8s;
`;

export const FlexRowContainer = styled(FlexColContainer)`
  flex-direction: row;
  justify-content: space-around;
  gap: 4em;
`;

export const UploadedImg = styled.img`
  width: 25%;
  max-height: 100%;
`;

export const H1 = styled.h1`
  font-family: ${(props) => props.theme.fontFamily.secondaryFont};
  font-size: 2.8rem;
  color: ${(props) => props.theme.colors.primaryUi};
  text-align: left;
  line-height: 130%;
`;

export const Span = styled.span`
  font-family: ${(props) => props.theme.fontFamily.secondaryFont};
  font-size: 2.8rem;
  color: #b1e0d9;
`;
