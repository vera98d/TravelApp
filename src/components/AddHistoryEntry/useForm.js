import { useState } from "react";
import validateForm from "./validateForm";
import historyEntriesService from "../../services/HistoryEntriesService";

function useForm(province) {
  const [values, setValues] = useState({
    title: "",
    city: "",
    description: "",
    image: null,
  });

  const [errors, setErrors] = useState({});
  const [loading, setLoading] = useState("notLoading");

  const handleChange = (e) => {
    const { name, value } = e.target;
    setValues({ ...values, [name]: value });
  };

  const handleImage = (file) => {
    setValues({ ...values, image: file });
  };

  const uploadImage = async () => {
    const data = new FormData();
    data.append("file", values.image);
    data.append("upload_preset", "memoryPictures");

    setLoading("loading");
    const res = await fetch(
      "https://api.cloudinary.com/v1_1/dagaqrq8j/image/upload",
      {
        method: "POST",
        body: data,
        "Content-Type": "application/json",
      }
    );
    setLoading("loaded");
    return res.json();
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    setErrors(validateForm(values));

    if (Object.keys(validateForm(values)).length === 0) {
      const response = await uploadImage();
      historyEntriesService.insert(
        values.title,
        values.city,
        values.description,
        response.secure_url,
        province
      );
    }
  };

  return {
    handleChange,
    values,
    handleSubmit,
    errors,
    handleImage,
    loading,
  };
}

export default useForm;
