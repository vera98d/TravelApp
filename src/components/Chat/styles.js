import styled from "styled-components";
import { scrollbarStyles } from "../../styles";

export const Container = styled.div`
  background-color: ${(props) => props.theme.colors.secondaryUi};
  height: 600px;
  border-radius: 15px;
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
`;

export const Header = styled.h3`
  text-align: center;
  background-color: ${(props) => props.theme.colors.primaryUi};
  border-radius: 12px 12px 0 0;
  padding: 15px;
  margin: 0;
  font-size: 30px;
  font-weight: 1000;
  font-family: ${(props) => props.theme.fontFamily.secondaryFont};
`;

export const MessagesList = styled.ul`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  flex-grow: 1;
  margin: 10px;
  padding: 0%;
  overflow: auto;
  ${scrollbarStyles}
`;

export const Footer = styled.div`
  padding: 20px;
  display: flex;
  align-items: center;
`;

export const Input = styled.input`
  flex-grow: 1;
  border: none;
  border-radius: 5px;
  border-style: none;
  margin-right: 10px;
  padding: 10px;
  outline: none;
`;

export const Button = styled.button`
  align-self: center;
  border-radius: 50%;
  border: none;
  height: 30px;
  width: 30px;
  background-color: ${(props) => props.theme.colors.primaryUi};
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  font-family: ${(props) => props.theme.fontFamily.secondaryFont};
`;

export const Message = styled.li`
  list-style: none;
  background-color: ${(props) => props.theme.colors.white};
  border-radius: ${(props) =>
    props.isOwn ? "10px 0 10px 10px" : "0 10px 10px 10px"};
  padding: 15px;
  margin: 5px;
  font-size: 18px;
  font-weight: 500;
  line-height: 22px;
  align-self: ${(props) => (props.isOwn ? "flex-end" : "flex-start")};
`;
