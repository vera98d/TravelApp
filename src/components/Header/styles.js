import styled from "styled-components";

export const Container = styled.div`
  background-color: ${(props) => props.theme.colors.headerBackground};
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-left: 15px;
  padding-right: 15px;
  width: 100%;
`;

export const Title = styled.h1`
  font-family: ${(props) => props.theme.fontFamily.primaryFont};
  font-size: 80px;
  font-weight: 500;
  margin-left: 80px;
  margin-right: 80px;
  padding: 40px 0 0 0;
  text-align: center;
  @media screen and (max-width: 1400px) {
    padding: 20px 0 0 0;
  }
  @media screen and (max-width: 1150px) {
    font-size: 60px;
    padding: 20px 0 0 0;
    line-height: 50px;
  }
  @media screen and (max-width: 560px) {
    width: 350px;
    overflow: hidden;
    text-overflow: ellipsis;
  }
`;

export const Subtitle = styled.h2`
  font-family: ${(props) => props.theme.fontFamily.secondaryFont};
  font-size: 50px;
  font-weight: 500;
  margin: 0;
  padding: 0 0 40px 0;
  text-align: center;
  @media screen and (max-width: 1150px) {
    font-size: 30px;
    padding: 0 0 20px 0;
  }
`;

export const Button = styled.button`
  position: absolute;
  display: flex;
  align-self: flex-start;
  align-items: center;
  justify-content: center;
  border-radius: 50%;
  border: none;
  height: 50px;
  width: 50px;
  background-color: ${(props) => props.theme.colors.tertiaryUi};
  top: 20px;
  left: 10px;
  padding: 0;
  cursor: pointer;
  svg {
    fill: white;
    width: 35px;
    height: 35px;
  }
  @media screen and (max-width: 1150px) {
    height: 40px;
    width: 40px;
    margin: 10px;
    top: 10px;
    svg {
      height: 25px;
      width: 25px;
    }
  }
`;

export const RightButton = styled(Button)`
  left: initial;
  right: 10px;
`;
