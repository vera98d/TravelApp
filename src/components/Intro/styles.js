import styled, { keyframes } from "styled-components";
import img from "../../../src/introImage.jpg";

export const Container = styled.div`
  background-image: url(${img});
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  width: 100vw;
  min-height: 100vh;
  filter: invert(25%);
`;

export const Title1 = styled.h1`
  margin: 0;
  font-family: ${(props) => props.theme.fontFamily.primaryFont};
  font-weight: 400;
  font-size: 300px;
  color: #260101;
  position: absolute;
  left: 70px;
  bottom: 200px;
`;

export const Title2 = styled.h1`
  margin: 0;
  font-family: ${(props) => props.theme.fontFamily.primaryFont};
  font-weight: 400;
  font-size: 220px;
  color: #260101;
  position: absolute;
  left: 80px;
  bottom: 0px;
`;

export const StartButton = styled.div`
  position: fixed;
  display: flex;
  flex-direction: row;
  bottom: 20%;
  right: 20%;
  align-items: center;
  justify-content: center;
  background-color: ${(props) => props.theme.colors.primaryUi};
  color: ${(props) => props.theme.colors.white};
  font-family: ${(props) => props.theme.fontFamily.secondaryFont};
  font-size: 25px;
  padding: 1em 2em;
  border-radius: 10px;
  cursor: pointer;
  transition: 0.3s;

  &:hover {
    background: #df9b89;
    animation-play-state: paused;
  }
`;

const translation = keyframes`
  0%{
      transform: translateX(15px);
  }
  10%{
    transform: translateX(0px);
  }
  20%{
      transform: translateX(15px);
  }
  30%{
    transform: translateX(0px);
  }
  
  100%{
    transform: translateX(0px);
  }
`;

export const Span1 = styled.span`
  color: white;
  display: block;
  margin-left: 1em;
  color: ${(props) => props.theme.colors.white};
  animation: ${translation} 1s 0.3s infinite both ease-in;
  transition: 0.5s;
`;
