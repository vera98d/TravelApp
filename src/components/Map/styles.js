import { MapContainer } from "react-leaflet";
import styled from "styled-components";
import gif from "../../../src/bike.gif";

export const Container = styled.div`
  display: flex;
  justify-content: center;
  padding: 40px 20px;
  background-image: url(${gif});
  background-size: 400px;
  background-repeat: no-repeat;
  background-position: bottom left;
`;

export const StyledMap = styled(MapContainer)`
  width: 600px;
  height: 600px;
  background-color: white;

  a {
    display: none;
  }
`;

export const DefaultProvince = {
  color: `#fff`,
  fillColor: "#BF7D65C4",
  fillOpacity: 1,
  opacity: 1.0,
};

export const HoveredProvince = {
  color: `#fff`,
  fillColor: "#DAE5E5",
  fillOpacity: 0.7,
};
