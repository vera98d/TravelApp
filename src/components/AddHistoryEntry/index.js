import React from "react";
import {
  Form,
  Input,
  TextArea,
  ButtonGroup,
  ErrorMessage,
  FlexColContainer,
  FlexRowContainer,
  Spinner,
  UploadedImg,
  H1,
  Span,
} from "./style";
import useForm from "./useForm";
import ImageInput from "../ImageInput";
import uploadImg from "./img/upload.svg";
import { PrimaryButton } from "../../styles";

function AddHistoryEntry(props) {
  const { handleChange, values, handleSubmit, errors, handleImage, loading } =
    useForm(props.province);

  if (loading === "loading") {
    return (
      <FlexColContainer align="center" gap="3em" width="100%">
        <H1>Please wait, we're saving your post...</H1>
        <Spinner />
      </FlexColContainer>
    );
  }
  if (loading === "loaded") {
    return (
      <FlexRowContainer>
        <UploadedImg src={uploadImg} />
        <FlexColContainer align="flex-start" gap="1.5em" width="60%">
          <H1>
            Your post{" "}
            <Span>
              {values.title} - {values.city}{" "}
            </Span>
            has been successfully saved!
          </H1>
        </FlexColContainer>
      </FlexRowContainer>
    );
  }

  if (loading === "notLoading") {
    return (
      <>
        <h2>New story</h2>
        <Form onSubmit={handleSubmit}>
          <Input
            id="title"
            name="title"
            type="text"
            value={values.title}
            onChange={handleChange}
            placeholder="Type title of your journey"
          ></Input>
          {errors.title && <ErrorMessage>{errors.title}</ErrorMessage>}

          <Input
            id="city"
            name="city"
            type="text"
            value={values.city}
            onChange={handleChange}
            placeholder="Type city of your journey"
          ></Input>
          {errors.city && <ErrorMessage>{errors.city}</ErrorMessage>}

          <TextArea
            id="description"
            name="description"
            value={values.description}
            onChange={handleChange}
            placeholder="Describe your story"
          ></TextArea>
          {errors.description && (
            <ErrorMessage>{errors.description}</ErrorMessage>
          )}

          <ButtonGroup>
            <ImageInput onChange={handleImage} />
            <PrimaryButton type="submit">Add story</PrimaryButton>
          </ButtonGroup>
          {errors.image && <ErrorMessage>{errors.image}</ErrorMessage>}
        </Form>
      </>
    );
  }
}

export default AddHistoryEntry;
