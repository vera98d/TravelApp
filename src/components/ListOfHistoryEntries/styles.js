import styled from "styled-components";
import { PrimaryButton, scrollbarStyles } from "../../styles";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
  height: 600px;
  max-height: 100%;
  border-radius: 12px;

  div {
    color: grey;
    font-family: ${(props) => props.theme.fontFamily.secondaryFont};
    font-size: 24px;
    padding: 150px 10px;
    height: 100%;
    text-align: center;

    p {
      span {
        text-transform: capitalize;
      }
    }
  }
`;

export const Header = styled.h3`
  font-size: 30px;
  font-weight: 1000;
  font-family: ${(props) => props.theme.fontFamily.secondaryFont};
  text-align: center;
  background-color: ${(props) => props.theme.colors.primaryUi};
  border-radius: 12px;
  padding: 15px;
  margin-bottom: 20px;
`;

export const EntriesList = styled.ul.attrs((props) => ({
  width: "20px",
  trackColor: props.theme.colors.thirdUi,
  thumbColor: props.theme.colors.fourthUi,
}))`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  flex-grow: 1;
  padding: 0%;
  overflow: auto;
  ${scrollbarStyles}
`;

export const Entry = styled.li`
  list-style: none;
  background-color: ${(props) => props.theme.colors.secondaryUi};
  font-family: ${(props) => props.theme.fontFamily.secondaryFont};
  font-size: 18px;
  font-style: italic;
  width: 90%;
  padding: 10px;
  margin: 5px 0;
  border: 2px solid transparent;
  border-radius: 10px;
  cursor: pointer;

  &:hover {
    border: 2px solid ${(props) => props.theme.colors.primaryUi};
    box-shadow: 0 2px 5px gray;
  }
`;
export const Button = styled(PrimaryButton)`
  align-self: flex-start;
  margin-top: 30px;
`;
