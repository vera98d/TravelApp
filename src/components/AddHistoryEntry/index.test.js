import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../styles";
import AddHistoryEntry from "../AddHistoryEntry";

test("is rendering expected elements", async () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <AddHistoryEntry />
    </ThemeProvider>
  );

  expect(container).toMatchSnapshot();
});

test("renders all elements", () => {
  render(
    <ThemeProvider theme={theme}>
      <AddHistoryEntry />
    </ThemeProvider>
  );

  const modalTitle = screen.getByText(/New story/i);
  const titleInput = screen.getByPlaceholderText(/Type title of your journey/i);
  const cityInput = screen.getByPlaceholderText(/Type city of your journey/i);
  const descriptionInput = screen.getByPlaceholderText(/Describe your story/i);
  const fileInput = screen.getByRole("img", {
    src: "./img/addImageButton.svg",
  });
  const submitButton = screen.getByRole("button", { type: "submit" });

  expect(modalTitle).toBeInTheDocument();
  expect(titleInput).toBeInTheDocument();
  expect(cityInput).toBeInTheDocument();
  expect(descriptionInput).toBeInTheDocument();
  expect(fileInput).toBeInTheDocument();
  expect(submitButton).toBeInTheDocument();
});

test("validates form properly", async () => {
  render(
    <ThemeProvider theme={theme}>
      <AddHistoryEntry />
    </ThemeProvider>
  );
  const submitButton = screen.getByRole("button", { type: "submit" });
  const titleInput = screen.getByPlaceholderText(/Type title of your journey/i);
  const cityInput = screen.getByPlaceholderText(/Type city of your journey/i);
  const descriptionInput = screen.getByPlaceholderText(/Describe your story/i);

  userEvent.click(submitButton);

  await waitFor(() => {
    expect(screen.getByText("Title is required.")).toBeInTheDocument();
    expect(screen.getByText("City is required.")).toBeInTheDocument();
    expect(screen.getByText("Description is required.")).toBeInTheDocument();
  });

  userEvent.type(titleInput, "Amazing adventure");
  userEvent.type(cityInput, "GdaÅ„sk");
  userEvent.type(
    descriptionInput,
    "It was a wonderful trip a had with my best friends"
  );
  userEvent.click(submitButton);

  await waitFor(() => {
    expect(screen.queryByText("Title is required.")).not.toBeInTheDocument();
    expect(screen.queryByText("City is required.")).not.toBeInTheDocument();
    expect(
      screen.queryByText("Description is required.")
    ).not.toBeInTheDocument();
  });
});
