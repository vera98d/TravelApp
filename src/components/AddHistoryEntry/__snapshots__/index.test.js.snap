// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is rendering expected elements 1`] = `
<div>
  <h2>
    New story
  </h2>
  <form
    class="sc-gsDKAQ bIehX"
  >
    <input
      class="sc-dkPtRN ddWviv"
      id="title"
      name="title"
      placeholder="Type title of your journey"
      type="text"
      value=""
    />
    <input
      class="sc-dkPtRN ddWviv"
      id="city"
      name="city"
      placeholder="Type city of your journey"
      type="text"
      value=""
    />
    <textarea
      class="sc-hKwDye fLiIyJ"
      id="description"
      name="description"
      placeholder="Describe your story"
    />
    <div
      class="sc-eCImPb eBCwcJ"
    >
      <label
        for="file-input"
      >
        <img
          class="sc-hBUSln JbMTg"
          src="addImageButton.svg"
        />
      </label>
      <input
        class="sc-bqiRlB UdnLw"
        id="file-input"
        type="file"
      />
      <button
        class="sc-bdvvtL eHPgEv"
        type="submit"
      >
        Add story
      </button>
    </div>
  </form>
</div>
`;
